infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationService.kt

```diff
@@ -1371,4 +1441,68 @@ class PublicationService @Autowired constructor(
         ratkoPushTime = if (publication.ratkoPushStatus == RatkoPushStatus.SUCCESSFUL) publication.ratkoPushTime else null,
         propChanges = propChanges,
     )
+
+    private fun enrichDuplicateNameExceptionOrRethrow(exception: DataIntegrityViolationException): Nothing {
```

Comment:
"Nothing" palautusarvon tyyppinä tarkoittaa, että funktiosta ei palauteta koskaan tavallista palautusarvoa. Sama kuin TypeScriptin "never". Tämä on se, mikä sallii käyttää tätä funktiota ylempänä publishChanges()in catch-blokissa.